### PR TITLE
Dockerfile: copy source instead of cloning the repo

### PR DESCRIPTION
**Pull request by 353fc443**
Instead of cloning the git repo copy the source from the PWD.

### CODE REVIEW

**Comment by JalonSolov** [comment]
I'm torn on this one.  On the one hand, it would be nice to be sure you have the latest V in the container, regardless if you have updated locally or not.  On the other hand, you might want to test local changes in a container.

Perhaps make it an option?  Perhaps by passing in an env var?  Let it default to cloning the repo, but allow a way to use the local source instead.

**Comment by 353fc443** [comment]
@JalonSolov Unfortunately conditional builds are not supported for docker.

**Comment by JalonSolov** [comment]
Actually, you can do it.  Check the `--build-arg` option, to pass values in, then you can have

RUN if [$arg == "foo"] then git clone ... else cp ...

types statements in the Dockerfile.

**Comment by 353fc443** [comment]
@JalonSolov   Is e563aca2cfe81bfaee4405f Okay?

**Comment by JalonSolov** [comment]
We really need to have similar changes in the other Dockerfile* files, and some documentation (otherwise, how will anyone know about this new option?), but that's for another PR.

Good job!

**Comment by 353fc443** [comment]
My bad I forgot to add `USE_LOCAL` Arg in the Dockerfile! fixed in c5b1f4a8848d24cdbf4fab36c67622dd1a35b3ab :)

File: Dockerfile.alpine [hunk 1]
```diff
@@ -10,14 +10,25 @@ ENV VFLAGS -cc gcc
...
   git make upx gcc bash \
   musl-dev \
   openssl-dev sqlite-dev \
   libx11-dev glfw-dev freetype-dev
 
 ## RUN apk --no-cache add --virtual sdl2deps sdl2-dev sdl2_ttf-dev sdl2_mixer-dev sdl2_image-dev
+COPY . /vlang-local
+
+RUN if [[ -z "${USE_LOCAL}" ]] ; then \
+      git clone https://github.com/vlang/v/ /opt/vlang && \
```
Comment (dumblob) [inline Dockerfile.alpine:25]: Maybe add `--depth 1` for speed?

**Comment by JalonSolov** [comment]
Probably not enough speed difference to be worthwhile.  It would only help speed up building the image, which is the `1` in the `1 to many` against the number of times you might run the image to do something.

**Pull request merged**